client/me/purchases/billing-history/billing-history-list.tsx

```diff
@@ -242,44 +261,109 @@ class BillingHistoryList extends Component {
 	};
 }
 
-function getIsSendingReceiptEmail( state ) {
-	return function isSendingBillingReceiptEmailForReceiptId( receiptId ) {
+function getIsSendingReceiptEmail( state: IAppState ) {
+	return function isSendingBillingReceiptEmailForReceiptId( receiptId: number ) {
 		return isSendingBillingReceiptEmail( state, receiptId );
 	};
 }
 
-BillingHistoryList.propTypes = {
-	//connected props
-	app: PropTypes.string,
-	date: PropTypes.object,
-	page: PropTypes.number,
-	pageSize: PropTypes.number.isRequired,
-	query: PropTypes.string.isRequired,
-	total: PropTypes.number.isRequired,
-	transactions: PropTypes.array,
-	//own props
-	siteId: PropTypes.number,
-	header: PropTypes.bool,
-};
+const PAGE_SIZE = 5;
+
+/**
+ * Utility function for formatting date for text search
+ */
+function formatDate( date: string ): string {
+	const localeSlug = getLocaleSlug();
+	return moment( date )
+		.locale( localeSlug ?? '' )
+		.format( 'll' );
+}
+
+/**
+ * Utility function extracting searchable strings from a single transaction
+ */
+function getSearchableStrings( transaction: BillingTransaction ): string[] {
+	const rootStrings: string[] = Object.values( transaction ).filter(
+		( value ) => typeof value === 'string'
+	);
+	const dateString: string | null = transaction.date ? formatDate( transaction.date ) : null;
+	const itemStrings: string[] = transaction.items.flatMap( ( item ) => Object.values( item ) );
+
+	return [ ...rootStrings, dateString, ...itemStrings ].filter( isValueTruthy );
+}
+
+/**
+ * Utility function to search the transactions by the provided searchQuery
+ */
+function searchTransactions(
+	transactions: BillingTransaction[],
+	searchQuery: string
+): BillingTransaction[] {
+	const needle = searchQuery.toLowerCase();
+
+	return transactions.filter( ( transaction: BillingTransaction ) =>
+		getSearchableStrings( transaction ).some( ( val ) => {
+			const haystack = val.toString().toLowerCase();
+			return haystack.includes( needle );
+		} )
+	);
+}
+
+function filterTransactions(
```

Comment:
This function and its dependencies were moved here from `get-billing-transaction-filters`, which was overly complex and only ever used here.